base_synchro/wizard/base_synchro.py

```diff
@@ -136,7 +136,14 @@ def synchronize(self, server, object):
                     pool = pool_dest.env[object.model_id.model]
                     pool.browse([id2]).write(value)
                 else:
-                    pool_dest.get(object.model_id.model).write([id2], value)
+                    try:
+                        pool_dest.get(object.model_id.model).write([id2],
+                                                                   value)
+                    except:
+                        raise Warning(_('''Synchronization is not be possible
+                        because some related records are either deleted or
+                        missing from destination database or from source
+                        database!'''))
```

Comment:
The sync should not be on HOLD due to one record. Log the warning in logger, and let the other ids continue.

Make the warning better:
'The synchronization will be skipped for the record %s, due to the relevant record is not in available in either of the DBs'